org.eclipse.cdt.cross.arc.gnu/src/com/arc/cdt/toolchain/tcf/TcfContent.java

```diff
@@ -152,7 +155,8 @@ public static TcfContent readFile(File f, String cpuFlag) throws TcfContentExcep
                 Element e = (Element) attributes.item(index);
                 String elementName = e.getAttribute("name");
                 if (!elementName.equals(GCC_OPTIONS_SECTION)
-                        && !elementName.equals(LINKER_MEMORY_MAP_SECTION)) {
+                        && !elementName.equals(LINKER_MEMORY_MAP_SECTION)
```

Comment:
I think that at this point you need to add a `static final List` of known sections and use `if (!knownSections.contains(elementName)) continue;`

Using list instead of HashSet makes sense because there are so few elements.